### PR TITLE
Fix DSN parser handling of bare string_quote token

### PR DESCRIPTION
**Pull request by CalyxNZ**
## Summary
- Handle SPECCTRA DSN parser declarations like `(string_quote ")` as a bare quote token instead of consuming the following file content as an unterminated quoted string.
- Make the debug-file helper path-safe on Windows and write files synchronously so image comparison tests can read generated output immediately.

## Context
This came up while checking #54 / the Smoothie Board conversion path. It fixes a parser/test blocker without changing the existing Smoothie repro behavior.

## Test plan
- `npx --yes bun test`
- `npx --yes bun run build`
- `./node_modules/.bin/biome format lib/common/parse-sexpr.ts tests/fixtures/get-test-debug-utils.ts`
- `git diff --check`

If this contribution was useful and you'd like to support the work, tips are welcome here: https://buymeacoffee.com/domenglands

/claim #54

Claim scope: this PR fixes the parser/test blocker described above while checking #54; it does not claim the full Smoothie conversion is complete.

### CODE REVIEW

**Comment by CalyxNZ** [comment]
Updated after review: bare `"` handling is now limited to a quote followed by `)` or EOF, so quoted strings that begin with whitespace are preserved. Re-ran `npx --yes bun test`, `npx --yes bun run build`, touched-file Biome format, and `git diff --check`.

File: lib/common/parse-sexpr.ts [hunk 1]
```diff
@@ -26,6 +26,12 @@ export function tokenizeDsn(input: string): Token[] {
       // Ignore whitespace
       i++
     } else if (char === '"') {
+      if (i + 1 >= length || /\s|\)/.test(input[i + 1])) {
+        tokens.push({ type: "String", value: '"' })
```
[the remark this answers is not in the document]
Comment (CalyxNZ) [reply lib/common/parse-sexpr.ts]: Thanks, good catch. I narrowed the bare quote handling to only `"` followed by `)` or EOF, so quoted values that begin with whitespace stay intact. Re-ran `npx --yes bun test`, `npx --yes bun run build`, touched-file Biome format, and `git diff --check`, then pushed the fix in `1bb5731`.

[the remark this answers is not in the document]
Comment (DomDigitalSimple) [reply lib/common/parse-sexpr.ts]: Added focused regression coverage in `31f8229` for both sides of this edge case: bare `(string_quote ")` tokenization and quoted strings that start with whitespace, e.g. `(net " GND" 1)`. Verification now passes with `npx --yes bun test` (49 pass, 1 skip), `npx --yes bun run build`, and `git diff --check`.

[the remark this answers is not in the document]
Comment (DomDigitalSimple) [reply lib/common/parse-sexpr.ts]: Tightened this one more step in `fac0549`: the bare quote case now also handles an optional whitespace separator before `)` or EOF, e.g. `(string_quote " )`, without treating quoted values that start with whitespace as bare tokens. Verification: `npx --yes bun test` (50 pass, 1 skip), `npx --yes bun run build`, and `git diff --check`.
